### ISSUE TITLE
Spotbugs ignores one particular `@SuppressFBWarnings` annotation.

### ISSUE THREAD

**Issue by McPringle** [spotbugs/spotbugs#1613]
Spotbugs ignores a suppress warnings annotation. But just one, other are working as expected.

Please take a look at my class `MainFrame.java` line 117:

```java
@SuppressFBWarnings(value = "DMI_RANDOM_USED_ONLY_ONCE", justification = "we need just ONE random number ONCE")
private float hue = new Random().nextFloat();
```

Why is Spotbugs ignoring the suppress warnings annotation? Others in the same project work as expected. I copied the value from the HTML report, so there should not be a typo.

Steps to reproduce:

```
git clone https://github.com/zdila/gpx-animator.git
cd gpx-animator
git checkout spotbugs-issue
./gradlew assemble
```

The build fails on Spotbugs problems. An HTML report can now be found in the build directory:

```
build/reports/checkstyle/main.html
```

I use Java 16.0.1 on macOS 11.4 with Spotbugs Gradle Plugin 4.7.1, Spotbugs 4.3.0, and Spotbugs Annotations 4.3.0.

Is the problem on my side? What have I overlooked?

**Comment by McPringle** [comment]
Sorry, my fault! It was too late! It was checkstyle, who was complaining, not spotbugs.

**Comment by ThrawnCA** [comment]
I don't think you can put the annotation on fields. Can you try initialising the field in the constructor instead, and put the annotation on that?

**Comment by slovdahl** [comment]
Is there a reason for the suppression annotation not working on fields or statements?

**Comment by iloveeclipse** [comment]
> I don't think you can put the annotation on fields.

See

https://github.com/spotbugs/spotbugs/blob/79ae7a26a57036c07b8dce4ffdc7d44a8e984c03/spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/SuppressFBWarnings.java#L25-L33

There is no `Target` annotation, which means, it **should** applicable to any element:

> If an @Target meta-annotation is not present on an annotation interface T, then an annotation of type T may be written as a modifier for any declaration.

https://docs.oracle.com/en/java/javase/25/docs/api//java.base/java/lang/annotation/Target.html

**Comment by slovdahl** [comment]
Yes, technically it's possible to annotate fields and statements, but it has no effect on SpotBugs. The suppression does not work, that is. Only suppressions on the class and method level work in reality.

**Comment by gtoison** [comment]
This happens because the suppress annotation is on the field but the bug is in the <init> method.
There's no notion that this bit of code is attached to the field, it just happens to be initialising it.
This should be fixed by #3897 by noticing that the problematic code is followed by a field put instruction.
